test/shell/test_diagnostics_reporter.sh

```diff
@@ -0,0 +1,164 @@
+# shellcheck source=./test_runner.sh
+dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "${dir}"/test_runner.sh
+. "${dir}"/test_helper.sh
+runner=$(get_test_runner "${1:-local}")
+
+export PATH="$(bazel info workspace)"/tools:$PATH
+
+build_target() {
+  set +e
+  target=$1
+  bazel build --build_event_publish_all_actions //test_expect_failure/diagnostics_reporter:"$target"
+  set -e
+}
+
+verify_output() {
+  target=$1
+  expected_outputs=$2
+  execution_root=$(bazel info execution_root)
+  output=$(protoc --decode_raw <"$execution_root"/bazel-out/darwin-fastbuild/bin/test_expect_failure/diagnostics_reporter/"$target".diagnosticsproto)
+  for expected_output in "${expected_outputs[*]}"; do
+    if [[ "$output" != *"$expected_output"* ]]; then
+      echo "Expected: $expected_output, got: $output"
+      exit 1
+    fi
+  done
+
+}
+
+verify_simple_file() {
+  build_target "simple_file"
+  outputs=("1 {
+  1: \"workspace-root://test_expect_failure/diagnostics_reporter/SimpleFile.scala\"
+  2 {
+    1 {
+      1 {
+        1: 5
+        2: 2
+      }
+      2 {
+        1: 6
+      }
+    }
+    2: 1
+    5: \"\')\' expected but \'}\' found.\"
```

Comment:
```
message Diagnostic {
  Range range = 1;
  Severity severity = 2;
  int64 code = 3;
  string message = 5;
  repeated DiagnosticRelatedInformation related_information = 6;
}
```
there are 5 fields in the struct, how come the output has some missing? maybe some comment may help.

Reply:
The fields you see there are defined as a diagnostic is defined in [LSP](https://microsoft.github.io/language-server-protocol/specification#diagnostic) (for context, BSP is an extension to LSP, thus sharing some information objects with it.). I don't think these are usable in the case of scala, though. Perhaps it would be interesting to leave a comment in the `proto` file explaining this, what do you think?

Reply:
Yeah it would be helpful to outline the differences.

LSP one has a few optional fields (is that what `?` stand for), is that why some aren't presented in this case?

Reply:
That's correct. I'll leave a note then!